mis_builder/models/mis_report_instance.py

```diff
@@ -133,6 +133,27 @@ def _compute_dates(self):
                 record.date_from = fields.Date.to_string(date_from)
                 record.date_to = fields.Date.to_string(date_to)
                 record.valid = True
+
+            elif record.mode == MODE_REL and record.type == "Q":  # Quarterly period logic
+                date_from = d.month(month=1)
+                date_from = date_from + relativedelta(months=record.offset * 3)
+                if d.month in [1, 2, 3]:
+                    date_from = d.replace(month=1, day=1)  # Q1
+                elif d.month in [4, 5, 6]:
+                    date_from = d.replace(month=4, day=1)  # Q2
+                elif d.month in [7, 8, 9]:
+                    date_from = d.replace(month=7, day=1)  # Q3
+                else:
+                    date_from = d.replace(month=10, day=1)  # Q4
```

Comment:
You could combine this to `date_from = replace(month=(datetime(2024, 7, 7).month - 1) % 3 + 1)`